urllib3/sessionmanager.py

```diff
@@ -0,0 +1,78 @@
+from .util.retry import Retry, handle_redirect
+from .request import RequestMethods, Request
+from .poolmanager import PoolManager, ProxyManager
+from .util.context import Context
+
+
+class SessionManager(RequestMethods):
+    """
+    Allows arbitrary requests while maintaining browser-session-like
+    context across those requests.
+
+    :param headers:
+        Headers to include with all requests, unless other headers are
+        given explcitly
+
+    :param cookie_jar:
+        A predefined :class:`CookieJar` object to use in the session;
+        if not provided, a new one will be created.
+
+    :param \**pm_kwargs:
+        Additional paramaters to be passed to the encapsulated
+        :class:`.PoolManager` or :class:`.ProxyManager` instance.
+    """
+    def __init__(self, headers=None, cookie_jar=None, **pm_kwargs):
+
+        super(SessionManager, self).__init__(headers=headers)
+
+        # Detect if we should create a PoolManager or a ProxyManager
+        if pm_kwargs.get('proxy_url', None) is None:
```

Comment:
I'm of two minds on this; on the one, a falsy string will pretty much never be a valid proxy. On the other, should urllib3 be expected to do anything to check validity on a high level when it receives a proxy URL, or just raise the appropriate exception when the connection fails? I'm fine either way, though, so I can make that change if you want.